kernel/src/lib.rs

```diff
@@ -533,6 +533,10 @@ pub trait StorageHandler: AsAny {
         &self,
         files: Vec<FileSlice>,
     ) -> DeltaResult<Box<dyn Iterator<Item = DeltaResult<Bytes>>>>;
+
+    /// Copy a file atomically from source to destination. If the destination file already exists,
+    /// it must return Err(Error::FileAlreadyExists).
+    fn copy(&self, src: &Url, dest: &Url) -> DeltaResult<()>;
```

Comment:
can we call this `copy_atomically` to make the semantics more clear

Reply:
what about `copy_atomic` for brevity?